helm/secdash/dags/update_mailchimp_list.py

```diff
@@ -6,161 +6,180 @@
 from projects import get_mongo_db
 
 
-def fetch_unique_emails(mongo_conn_id):
-    try:
-        db = get_mongo_db(mongo_conn_id)
-        projects_collection = db["PrivateCloudProject"]
-        users_collection = db["User"]
-
-        projects = projects_collection.find({"status": "ACTIVE"})
-
-        unique_emails = set()
-
-        for project in projects:
-            project_owner_id = project.get("projectOwnerId")
-            if project_owner_id:
-                project_owner = users_collection.find_one({"_id": ObjectId(project_owner_id)})
-                if project_owner:
-                    email = project_owner.get('email')
-                    if email:
-                        unique_emails.add(email)
-
-            primary_technical_lead_id = project.get("primaryTechnicalLeadId")
-            if primary_technical_lead_id:
-                primary_technical_lead = users_collection.find_one({"_id": ObjectId(primary_technical_lead_id)})
-                if primary_technical_lead:
-                    email = primary_technical_lead.get("email")
-                    if email:
-                        unique_emails.add(email)
-
-            secondary_technical_lead_id = project.get("secondaryTechnicalLeadId")
-            if secondary_technical_lead_id:
-                secondary_technical_lead = users_collection.find_one({"_id": ObjectId(secondary_technical_lead_id)})
-                if secondary_technical_lead:
-                    email = secondary_technical_lead.get("email")
-                    if email:
-                        unique_emails.add(email)
-
-        return list(unique_emails)
-
-    except Exception as e:
-        print(f"[fetch_unique_emails] Error: {e}")
-        return []
-
-
-def add_emails_to_list(client, list_id, emails):
-    try:
+class MailchimpManager:
+    def __init__(self, api_key, server_prefix, list_id, tag_id, mongo_conn_id):
+        self.client = MailchimpMarketing.Client()
+        self.client.set_config({
+            "api_key": api_key,
+            "server": server_prefix
+        })
+        self.list_id = list_id
+        self.db = get_mongo_db(mongo_conn_id)
+        self.tag_id = tag_id
+
+    def get_email_hash(self, email):
+        return hashlib.md5(email.lower().encode('utf-8')).hexdigest()
+
+    def fetch_unique_emails(self):
```

Comment:
Since this function is generic and not tied to the context of `mailchimp`, let's move it out of the class method.